From b847c2fe31693ddb7395b6f1de91f7d87ec9ba9c Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Sat, 22 Jun 2013 00:35:31 +0530
Subject: [PORT FROM MAIN]Music still output from IHF when plug in HDMI cable

BZ: 84245 79682

This patch will set offload tear down flag when HDMI or BT
connected during offload playback.

When HDMI or BT is connected during music offload playback, offload
playback will tear down and start playback with IA decoding.

Change-Id: I1ffeac5c13a6c077198a1970bf2e23cbd1192519
Orig-Change-Id: I791ce00583ac629812ec0f0ff1f342a3a9037412
Signed-off-by: samaga Prasanna Krishna <samaga.krishna@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 services/audioflinger/Threads.cpp |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/services/audioflinger/Threads.cpp b/services/audioflinger/Threads.cpp
index fd42ef3..b412af5 100644
--- a/services/audioflinger/Threads.cpp
+++ b/services/audioflinger/Threads.cpp
@@ -1800,6 +1800,12 @@ void AudioFlinger::PlaybackThread::invalidateTracks(audio_stream_type_t streamTy
     for (size_t i = 0; i < size; i++) {
         sp<Track> t = mTracks[i];
         if (t->streamType() == streamType) {
+#ifdef INTEL_MUSIC_OFFLOAD_FEATURE
+            if ((t->isOffloaded()) && (!t->isPaused()) &&
+                (!(t->mCblk->flags & CBLK_OFFLOAD_TEAR_DOWN_ON))) {
+                android_atomic_or(CBLK_OFFLOAD_TEAR_DOWN_ON, &t->mCblk->flags);
+            }
+#endif
             t->invalidate();
         }
     }
-- 
1.7.4.1

